<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.433">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Joseph Womersley">

<title>Supplementary Data Analysis Methods</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="Supplementary_Data_Analysis_Methods_files/libs/clipboard/clipboard.min.js"></script>
<script src="Supplementary_Data_Analysis_Methods_files/libs/quarto-html/quarto.js"></script>
<script src="Supplementary_Data_Analysis_Methods_files/libs/quarto-html/popper.min.js"></script>
<script src="Supplementary_Data_Analysis_Methods_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="Supplementary_Data_Analysis_Methods_files/libs/quarto-html/anchor.min.js"></script>
<link href="Supplementary_Data_Analysis_Methods_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="Supplementary_Data_Analysis_Methods_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="Supplementary_Data_Analysis_Methods_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="Supplementary_Data_Analysis_Methods_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="Supplementary_Data_Analysis_Methods_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Supplementary Data Analysis Methods</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Joseph Womersley </p>
          </div>
  </div>
    
  
    
  </div>
  

</header>

<section id="data-description" class="level2">
<h2 class="anchored" data-anchor-id="data-description">Data Description</h2>
<p>The data files used in this report: <a href="https://joewomersley.github.io/report_quarto/hspc_surfaceome.html">surfaceome_hspc.csv</a>, <a href="https://joewomersley.github.io/report_quarto/lthsc_surfaceome.html">surfaceome_lthsc.csv</a>, <a href="https://joewomersley.github.io/report_quarto/prog_surfaceome.html">surfaceome_prog.csv</a>, <a href="https://joewomersley.github.io/report_quarto/er_cell_types.html">er_cell_types.csv</a> and <a href="https://joewomersley.github.io/report_quarto/qPCR_raw_data.html">qPCR_raw_data.xlsx</a></p>
<p>The surfaceome data contains Log<sub>2</sub> absolute expression values. Each row is a gene represented by its ensembl gene id. Also in the files are columns for each specific cell labelled numerically.</p>
<p>The data in <a href="https://joewomersley.github.io/report_quarto/er_cell_types.html">er_cell_types.csv</a> redefines the previous cell type definitions. Each row is an old cell definition for all of the three cell types. Each column represents a new cell type definition whereby going down the column, a <strong><code>1</code></strong> indicates that new cell definition can be used to re-define the old cell type, a <strong><code>0</code></strong> indicates it does not.</p>
<p>The data in <a href="https://joewomersley.github.io/report_quarto/qPCR_raw_data.html">qPCR_raw_data</a> contains relative abundance values of the <em>Mmrn1</em> gene. Each row is a cell and its relative abundance with the columns being the cell_type, followed by <em>Mmrn1</em> mean expression. The mean expression is calculated by using expression of <em>Mmrn1</em> done in triplicate and normalized against <em>Hprt</em> to determine its relative abundance.</p>
</section>
<section id="data-preparation" class="level2">
<h2 class="anchored" data-anchor-id="data-preparation">Data Preparation</h2>
<p>We used R (<a href="https://www.r-project.org/">R Core Team 2021</a>) with.<strong><code>tidyverse</code></strong>.packages<span class="citation" data-cites="wickham2019"><sup><a href="#ref-wickham2019" role="doc-biblioref">1</a></sup></span> to import and process the raw data. Differential expression analysis was then performed to determine the log fold change between cell types. We used the .<strong><code>findmarkers()</code></strong>. function to run this differential analysis within the .<strong><code>scran</code></strong>. package<span class="citation" data-cites="lun2016"><sup><a href="#ref-lun2016" role="doc-biblioref">2</a></sup></span> which also employs statistical testing upon the analysis to generate p.values and false discovery rate.</p>
</section>
<section id="visualisation-through-mean-expression-plots-with-error-bars-and-histograms" class="level2">
<h2 class="anchored" data-anchor-id="visualisation-through-mean-expression-plots-with-error-bars-and-histograms">Visualisation through Mean Expression Plots with Error Bars and Histograms</h2>
<p>Mean expression plots with error bars and histograms were used to visualise the gene expression across the cells and genes to ensure they are similar across the cell types, and also to ensure that there were minimal outliers, as well as to further understand the distribution in expression. The .<strong><code>tidyverse</code></strong>.packages<span class="citation" data-cites="wickham2019"><sup><a href="#ref-wickham2019" role="doc-biblioref">1</a></sup></span> were used to generate the plots.</p>
</section>
<section id="volcano-plots" class="level2">
<h2 class="anchored" data-anchor-id="volcano-plots">Volcano Plots</h2>
<p>Volcano plots were then used to visualise the differential analysis through using the .<strong><code>ggplot2</code></strong>. package<span class="citation" data-cites="ggplot2"><sup><a href="#ref-ggplot2" role="doc-biblioref">3</a></sup></span> whereby we coloured points with &nbsp;q&nbsp;&lt; 0.05 where&nbsp;q&nbsp;is the False Discovery Rate (FDR) adjusted&nbsp;p-value. FDR is a method of correcting for multiple comparisons by the Benjamini–Hochberg procedure<span class="citation" data-cites="benjamini1995"><sup><a href="#ref-benjamini1995" role="doc-biblioref">4</a></sup></span>. Additionally, we labelled points with their corresponding gene name, when their Log fold change was greater than 2.5. We did this using the .<strong><code>BiomaRt</code></strong>. package<span class="citation" data-cites="biomaRt"><sup><a href="#ref-biomaRt" role="doc-biblioref">5</a></sup></span> and created a logical variable .<strong><code>bigfc</code></strong>. which represents whether the absolute value of the log fold change for a particular gene exceeds 2.5, and used the .<strong><code>ggrepel</code></strong>. package to avoid labels overlapping.</p>
</section>
<section id="bar-plots" class="level2">
<h2 class="anchored" data-anchor-id="bar-plots">Bar Plots</h2>
<p>All bar plots were produced using .<strong><code>tidyverse</code></strong>.packages<span class="citation" data-cites="wickham2019"><sup><a href="#ref-wickham2019" role="doc-biblioref">1</a></sup></span>, specifically the .<strong><code>ggplot2</code></strong>. package<span class="citation" data-cites="ggplot2"><sup><a href="#ref-ggplot2" role="doc-biblioref">3</a></sup></span>. Whereby all bars have their standard error also visualised, Error bars are&nbsp;x¯±1s.e.&nbsp;where&nbsp;s.e.&nbsp;is the standard error of the mean. P values: <em>*, P &lt; 0.05; **</em>, P &lt; 0.01; ***, P &lt; 0.001; ****, P &lt; 0.0001.</p>
</section>
<section id="tables" class="level2">
<h2 class="anchored" data-anchor-id="tables">Tables</h2>
<p>Tables were generated to show differentially expressed genes (logFC &gt; 1) for each of the three comparisons, both with and without duplicate genes showing different comparisons. The .<strong><code>knitr</code></strong>. package<span class="citation" data-cites="knitr"><sup><a href="#ref-knitr" role="doc-biblioref">6</a></sup></span> was used to enable the analysis and the .<strong><code>KableExtra</code></strong>. package<span class="citation" data-cites="kableExtra"><sup><a href="#ref-kableExtra" role="doc-biblioref">7</a></sup></span> was used to format the tables.</p>
</section>
<section id="visualisation-with-principal-components-analysis" class="level2">
<h2 class="anchored" data-anchor-id="visualisation-with-principal-components-analysis">Visualisation With Principal Components Analysis</h2>
<p>Dimensionality-reduction was performed on the Log<sub>2</sub>-transposed absolute expression data of the 280 genes by using principle component analysis. The Package .<strong><code>Plotly</code></strong>.<span class="citation" data-cites="plotly"><sup><a href="#ref-plotly" role="doc-biblioref">8</a></sup></span> was used to visualise the first 3 principle components, generating a three-dimensional PCA plot.</p>
</section>
<section id="heatmaps-and-clustering" class="level2">
<h2 class="anchored" data-anchor-id="heatmaps-and-clustering">Heatmaps And Clustering</h2>
<p>Hierarchical clustering was performed and heatmaps produced with the package .<strong><code>Pheatmap</code></strong>.<span class="citation" data-cites="pheatmap"><sup><a href="#ref-pheatmap" role="doc-biblioref">9</a></sup></span> with .<strong><code>BiomaRt</code></strong>.<span class="citation" data-cites="biomaRt"><sup><a href="#ref-biomaRt" role="doc-biblioref">5</a></sup></span> used to convert the ensembl gene ids to gene names. Hierarchical clustering used the Ward.D2 method and was performed for both genes and cells. For the heatmaps, only the top 50 significantly differentially expressed genes were visualised. Genes were deemed significantly differentially expressed when LogFC &gt; 1, which left 101 genes; we took the top 50 of these genes and deemed them significant by &nbsp;q&nbsp;&lt; 0.01.</p>

</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body" data-line-spacing="2" role="list">
<div id="ref-wickham2019" class="csl-entry" role="listitem">
<div class="csl-left-margin">1. </div><div class="csl-right-inline">Wickham, H. <em>et al.</em> <a href="https://doi.org/10.21105/joss.01686">Welcome to the tidyverse</a>. <em>Journal of Open Source Software</em> <strong>4</strong>, 1686 (2019).</div>
</div>
<div id="ref-lun2016" class="csl-entry" role="listitem">
<div class="csl-left-margin">2. </div><div class="csl-right-inline">Lun, A. T. L., McCarthy, D. J. &amp; Marioni, J. C. <a href="https://doi.org/10.12688/f1000research.9501.2">A step-by-step workflow for low-level analysis of single-cell RNA-seq data with Bioconductor</a>. <em>F1000Research</em> <strong>5</strong>, 2122 (2016).</div>
</div>
<div id="ref-ggplot2" class="csl-entry" role="listitem">
<div class="csl-left-margin">3. </div><div class="csl-right-inline">Wickham, H. <a href="https://ggplot2.tidyverse.org">ggplot2: Elegant graphics for data analysis</a>. (2016).</div>
</div>
<div id="ref-benjamini1995" class="csl-entry" role="listitem">
<div class="csl-left-margin">4. </div><div class="csl-right-inline">Benjamini, Y. &amp; Hochberg, Y. <a href="https://www.jstor.org/stable/2346101">Controlling the false discovery rate: A practical and powerful approach to multiple testing</a>. <em>Journal of the Royal Statistical Society. Series B (Methodological)</em> <strong>57</strong>, 289–300 (1995).</div>
</div>
<div id="ref-biomaRt" class="csl-entry" role="listitem">
<div class="csl-left-margin">5. </div><div class="csl-right-inline">Durinck, S., Spellman, P. T., Birney, E. &amp; Huber, W. Mapping identifiers for the integration of genomic datasets with the r/bioconductor package biomaRt. <strong>4</strong>, (2009).</div>
</div>
<div id="ref-knitr" class="csl-entry" role="listitem">
<div class="csl-left-margin">6. </div><div class="csl-right-inline">Xie, Y. <a href="https://yihui.org/knitr/">Knitr: A general-purpose package for dynamic report generation in r</a>. (2023).</div>
</div>
<div id="ref-kableExtra" class="csl-entry" role="listitem">
<div class="csl-left-margin">7. </div><div class="csl-right-inline">Zhu, H. <a href="https://CRAN.R-project.org/package=kableExtra">kableExtra: Construct complex table with ’kable’ and pipe syntax</a>. (2024).</div>
</div>
<div id="ref-plotly" class="csl-entry" role="listitem">
<div class="csl-left-margin">8. </div><div class="csl-right-inline">Sievert, C. <a href="https://plotly-r.com">Interactive web-based data visualization with r, plotly, and shiny</a>. (2020).</div>
</div>
<div id="ref-pheatmap" class="csl-entry" role="listitem">
<div class="csl-left-margin">9. </div><div class="csl-right-inline">Kolde, R. <a href="https://CRAN.R-project.org/package=pheatmap">Pheatmap: Pretty heatmaps</a>. (2019).</div>
</div>
</div></section></div></main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>